examples/with-nextjs-and-clerk-auth/src/components/NavigationMenu/NavigationListItem.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { FC, PropsWithChildren, ReactNode } from 'react';
+
+import { usePathname } from 'next/navigation';
+
+import {
+  ListItem,
+  ListItemButton,
+  ListItemButtonProps,
+  ListItemIcon,
+  ListItemText,
+  Typography,
+} from '@mui/material';
+
+type NavigationLinkProps = PropsWithChildren<{
+  endIcon?: ReactNode;
+  href?: string;
+  icon: ReactNode;
+  onClick?: ListItemButtonProps['onClick'];
+}>;
+
+export const NavigationListItem: FC<NavigationLinkProps> = (props) => {
+  const { endIcon, children, href, icon, onClick } = props;
+
+  const pathname = usePathname();
+  const selected = pathname === href?.replace(window.location.origin, '');
+
+  const buttonProps =
+    href === undefined
+      ? { component: 'button', onClick }
+      : { component: 'a', href };
+
+  return (
+    <ListItem className="navigation-list-item">
+      <ListItemButton
```

Comment:
We must use `component={Link}` from the `import Link from 'next/link';` to be compatible with Next.js routing.